pkgs/by-name/pdfding/package.nix

```
@@ -0,0 +1,194 @@
+    mv pdfding/core/settings/dev.py pdfding/core/settings/dev.py.bak
+
+    ${python.pythonOnBuildForHost.interpreter} pdfding/manage.py collectstatic
+
+    # dev.py is required so that test will run properly, restore it
+    mv pdfding/core/settings/dev.py.bak pdfding/core/settings/dev.py
+
+    # not needed, now we have staticfiles directory
+    rm -rf pdfding/static
+
+    # remove django md5 hash from filenames of pdfjs as it will mess up the relative imports because of the whitenoise setup
+    sh -x \
+      && export PDFJS_PATH="pdfding/staticfiles/pdfjs" \
+      && for file_name in $(find $PDFJS_PATH -type f -not -path "$PDFJS_PATH/web/images/*");  \
+        do \
+          if [[ $file_name =~ "LICENSE" ]]; then \
+            new=$(echo "$file_name" | sed -E "s/LICENSE\.[a-zA-Z0-9]{12}/LICENSE/"); \
+          else \
+            new=$(echo "$file_name" | sed -E "s/\.[a-zA-Z0-9]{12}\./\./"); \
+          fi; \
+          mv -- "$file_name" "$new"; \
+        done \
+      && echo 'Successfully removed hash from pdfjs files'
+
+    echo "VERSION = '${version}'" > pdfding/core/settings/version.py;
+  '';
+
+  postInstall = ''
+    mkdir -p $out/{bin,share}
+    pdfdingDir=$out/${python.sitePackages}/pdfding
+
+    # make an empty dir to supress the warning
+    mkdir -p $pdfdingDir/static
+
+    makeWrapper "$pdfdingDir/manage.py" $out/bin/pdfding-manage \
+      --set-default DATA_DIR "/var/lib/pdfding" \
+      --prefix PYTHONPATH : "${pythonPath}"
+
+    makeWrapper ${lib.getExe python.pkgs.gunicorn} $out/bin/pdfding-start \
+      --set-default DATA_DIR "/var/lib/pdfding" \
+      --prefix PYTHONPATH : "${pythonPath}:$pdfdingDir" \
+      --add-flags '--bind $HOST_IP:$HOST_PORT core.wsgi:application'
+  '';
+
+  pythonRelaxDeps = true;
+
+  nativeCheckInputs = with python.pkgs; [
+    pytest-cov-stub
+    pytest-django
+    pytestCheckHook
+  ];
+
+  # from .github/workflows/tests.yaml
+  pytestFlags = [
+    "--ignore=e2e"
+    "--cov=admin"
+    "--cov=backup"
+    "--cov=base"
+    "--cov=pdf"
+    "--cov=users"
+    "--cov-fail-under=100"
+  ];
+
+  /*
+    fix two breaking tests by providing full out path
+    AssertionError: Calls not found
+    AssertionError: 'add_file_to_minio' does not contain all of ...
+  */
+  preCheck = ''
+    pushd pdfding || exit 1
+    substituteInPlace backup/tests/test_management.py backup/tests/test_tasks.py \
+      --replace-fail "Path(__file__).parents[2]" "Path('$out/${python.sitePackages}/pdfding')"
+  '';
+
+  postCheck = ''
+    popd || exit 1
+  '';
+
+  # dev.py is required for tests and MUST be removed from the final output
+  # this could be done in postCheck, but doing it here will allow doCheck to be toggleable
+  postPhases = [ "finalPhase" ];
+
+  finalPhase = ''
+    # dev.py should be removed on production build (source Dockerfile)
+    # can't be removed earlier, required for checkPhase
+    rm $out/${python.sitePackages}/pdfding/core/settings/dev.py
+  '';
```

[Fixup phases](https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#the-fixup-phase-ssec-fixup-phase) are precisely made for this purpose.

> The fixup phase performs (Nix-specific) post-processing actions on the files installed under `$out` by the install phase.

```suggestion
  postFixup = ''
    # dev.py should be removed on production build (source Dockerfile)
    rm $out/${python.sitePackages}/pdfding/core/settings/dev.py
  '';
```